lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -380,12 +380,19 @@ llvm::json::Value DAP::CreateTopLevelScopes() {
   return llvm::json::Value(std::move(scopes));
 }
 
-ExpressionContext DAP::DetectExpressionContext(lldb::SBFrame &frame,
-                                               std::string &text) {
+static std::string FirstTerm(llvm::StringRef input) {
```

Comment:
better return `std::optional<llvm::StringRef>`

Reply:
I ended up re-writing this to inline the helper.